routing/index_graph_starter.cpp

```diff
@@ -4,190 +4,98 @@
 
 namespace routing
 {
-IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, RoadPoint const & startPoint,
-                                     RoadPoint const & finishPoint)
-  : m_graph(graph)
-  , m_start(graph, startPoint, graph.GetNumJoints())
-  , m_finish(graph, finishPoint, graph.GetNumJoints() + 1)
+IndexGraphStarter::IndexGraphStarter(IndexGraph & graph, FakeVertex const & start,
+                                     FakeVertex const & finish)
+  : m_graph(graph), m_start(start, false /* backward */), m_finish(finish, true /* forward */)
 {
-  m_start.SetupJointId(graph);
-
-  if (startPoint == finishPoint)
-    m_finish.m_jointId = m_start.m_jointId;
-  else
-    m_finish.SetupJointId(graph);
 }
 
-m2::PointD const & IndexGraphStarter::GetPoint(Joint::Id jointId)
+m2::PointD const & IndexGraphStarter::GetPoint(Segment const & segment, bool isOutgoing)
 {
-  if (jointId == m_start.m_fakeId)
-    return m_graph.GetGeometry().GetPoint(m_start.m_point);
+  if (segment == m_start.GetFakeSegment())
+    return m_start.GetPoint();
```

Comment:
Do we need to check isOutgoing here too? I guess, it's possible as m_start and m_finish were build from nearest edges.